src/librustc_metadata/locator.rs

```diff
@@ -889,6 +889,24 @@ fn get_metadata_section_imp(target: &Target,
         let blob = MetadataBlob::Raw(buf);
         verify_decompressed_encoding_version(&blob, filename)?;
         return Ok(blob);
+    } else if target.options.is_like_msvc {
+        // Treat the import library as though it were an rlib!
+        let ref implib_filename = filename.with_extension("dll.rlib");
```

Comment:
This looks basically equivalent to the rlib branch above, can these branches be shared?